shlr/sdb/src/ht_uu.c

```diff
@@ -1,19 +1,112 @@
 /* sdb - MIT - Copyright 2018-2022 - ret2libc, pancake */
 
-#include "sdb/sdb.h"
+#include <stdint.h>
 #include "sdb/ht_uu.h"
-#include "ht.inc.c"
-
-// creates a default HtUU that has strings as keys
-SDB_API HtName_(Ht)* Ht_(new0)(void) {
-	HT_(Options) opt = {
-		.cmp = NULL,
-		.hashfn = NULL,
-		.dupkey = NULL,
-		.dupvalue = NULL,
-		.calcsizeK = NULL,
-		.calcsizeV = NULL,
-		.freefn = NULL
-	};
-	return Ht_(new_opt) (&opt);
+#include "sdb/heap.h"
+#include "../../../libr/include/cwisstable.h"  // XXX use cwisstable from sdb itself
+
+typedef uint64_t ut64;
+
+#if defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wtypedef-redefinition"
+#endif
+CWISS_DECLARE_FLAT_HASHMAP(HtUU, ut64, ut64);
+#if defined(__GNUC__)
+#pragma GCC diagnostic pop
+#endif
+
+SDB_API HtUU* ht_uu_new0(void) {
+	HtUU *hm = sdb_gh_calloc (1, sizeof (HtUU));
+	assert (hm);
+	*hm = HtUU_new (0);
+	return hm;
+}
+
+SDB_API void ht_uu_free(HtUU *hm) {
+	if (hm) {
+		HtUU_destroy (hm);
+		free (hm);
+	}
+}
+
+SDB_API bool ht_uu_insert(HtUU *hm, const ut64 key, ut64 value) {
+	assert (hm);
+
+	HtUU_Entry entry = { .key = key, .val = value };
+	HtUU_Insert result = HtUU_insert (hm, &entry);
+	return result.inserted;
+}
+
+SDB_API bool ht_uu_update(HtUU *hm, const ut64 key, ut64 value) {
+	assert (hm);
```

Comment:
dont use assert. libraries shuold never crash. use r_return_val_if_fail... but those macros shuold be moved into sdb and used there.

also we differentiate from runtime and compile time asserts. this iis a compile time chk because by running the tests we can find if this function was called with null somwehre.

Reply:
Yeah I know, only used assert because that macro wasn't available there yet.